includes/wf_crm_webform_postprocess.inc

```diff
@@ -2225,16 +2225,22 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
    * @param $table
    * @param $n
    * @param $name
+   * @param $checkSubmitDisabledSetting
+   *   Checks if 'Submit Disabled' setting should be considered
    * @return bool
    */
-  private function isFieldHiddenByExistingContactSettings($ent, $c, $table, $n, $name) {
+  private function isFieldHiddenByExistingContactSettings($ent, $c, $table, $n, $name, $checkSubmitDisabledSetting = FALSE) {
     if ($ent == 'contact' && isset($this->enabled["civicrm_{$c}_contact_1_contact_existing"])) {
       $component = $this->getComponent("civicrm_{$c}_contact_1_contact_existing");
       $existing_contact_val = $this->submissionValue($component['cid']);
       // Fields are hidden if value is empty (no selection) or a numeric contact id
       if (!$existing_contact_val[0] || is_numeric($existing_contact_val[0])) {
         $type = ($table == 'contact' && strpos($name, 'name')) ? 'name' : $table;
         if (in_array($type, $component['extra']['hide_fields'])) {
+          // Check if 'Submit Disabled' setting is switched on and do it for contact_2 only as the data for contact 2 doesn't get to the webform submissions
+          if ($checkSubmitDisabledSetting && $component['extra']['submit_disabled']) {
```

Comment:
Update comment to remove references to contact 2 -> to something like: "Check to see if configured to 'Submit disabled field value(s)"